clippy_utils/src/consts.rs

```diff
@@ -302,6 +303,8 @@ pub enum ConstantSource {
     Local,
     /// The value is dependent on a defined constant.
     Constant,
+    /// The value is dependent on a constant defined in current crate of `core` crate.
+    LocalOrCoreCrateConstant,
```

Comment:
Let's keep it just core constants, local ones may have "arbitrary" values just as remote ones do while the ones in core are pretty straight forward